File: read_excel.js
const { execSync } = require('child_process');
const path = require('path');
const fs = require('fs');

// Try to find python and openpyxl
try {
  const result = execSync('where python 2>&1 || where python3 2>&1 || echo "NOT_FOUND"', { encoding: 'utf8' });
  console.log('which python:', result.trim());
} catch(e) {}

// Look for venv
const base = 'C:\\Users\\Utente\\Desktop\\documenti elia (appunti)\\fantacalcio\\fanta27\\coach-beard-starter 2\\coach-beard-starter';
const candidates = [
  path.join(base, '.venv', 'Scripts', 'python.exe'),
  path.join(base, 'venv', 'Scripts', 'python.exe'),
  path.join(base, '.env', 'Scripts', 'python.exe'),
];
for (const c of candidates) {
  console.log(c, fs.existsSync(c) ? 'EXISTS' : 'missing');
}

// Check node packages
console.log('\nnode_modules root:', fs.readdirSync(path.join(base, 'node_modules')).slice(0,20));

// Try openpyxl via python
try {
  const py = execSync('python -c "import openpyxl; print(openpyxl.__version__)" 2>&1', { encoding: 'utf8' });
  console.log('openpyxl version:', py.trim());
} catch(e) {
  console.log('openpyxl check failed:', e.message?.slice(0,200));
}

File: read_excel.py
import openpyxl
import os

base = r"C:\Users\Utente\Desktop\documenti elia (appunti)\fantacalcio\fanta27\coach-beard-starter 2\coach-beard-starter"

files = [
    os.path.join(base, 'Dati/Dati Su Giocatori, Allenatori e Squadre/Guida_Asta_202627.xlsx'),
    os.path.join(base, 'Dati/Listone E quotazioni/Quotazioni_Fantacalcio_Stagione_2026_27.xlsx'),
    os.path.join(base, 'Dati/Strategie/StrategiaFanta.xlsx'),
    os.path.join(base, 'Dati/Strategie/Pupilli.xlsx'),
]

for f in files:
    print('='*90)
    print('FILE:', os.path.relpath(f, base))
    try:
        wb = openpyxl.load_workbook(f, data_only=True)
        print('SHEETS:', wb.sheetnames)
        for sn in wb.sheetnames:
            ws = wb[sn]
            print(f'  Sheet [{sn}]: {ws.max_row} rows x {ws.max_column} cols')
            for i, row in enumerate(ws.iter_rows(values_only=True)):
                if i >= 6: break
                print('   ROW', i, ':', row)
            print()
    except Exception as e:
        print('  ERROR:', type(e).__name__, e)
